### PR TITLE
Add runtime type extraction to the UDx

### PR DESCRIPTION
**Pull request by bluescarni**

### CODE REVIEW

File: doc/sphinx/install.rst [hunk 1]
```diff
@@ -60,6 +60,7 @@ to the channels, and then we can immediately install pagmo:
 .. code-block:: console
 
    $ conda config --add channels conda-forge
+   $ conda config --set channel_priority strict
```
Comment (darioizzo) [inline doc/sphinx/install.rst]: out of curiosity why was this needed?

File: include/pagmo/algorithm.hpp [hunk 2]
```diff
@@ -640,6 +668,58 @@ class PAGMO_DLL_PUBLIC algorithm
...
+     * to the internal UDA instance. Differently from
+     * :cpp:func:`~pagmo::algorithm::extract()`, this function
+     * does not require to pass the correct type
+     * in input. It is however the user's responsibility
+     * to cast the returned void pointer to the correct type.
+     *
+     * .. note::
+     *
+     *    The returned value is a raw non-owning pointer: the lifetime of the pointee is tied to the lifetime
+     *    of ``this``, and ``delete`` must never be called on the pointer.
```
Comment (darioizzo) [inline include/pagmo/algorithm.hpp]: What happens if delete IS called? Can you just add a brief explanation?

File: include/pagmo/rng.hpp [hunk 3]
```diff
@@ -57,7 +57,11 @@ using random_engine_type = std::mt19937;
  */
 struct PAGMO_DLL_PUBLIC random_device {
     static unsigned next();
-    static void set_seed(unsigned);
+
+#if !defined(PAGMO_DOXYGEN_INVOKED)
```
Comment (darioizzo) [inline include/pagmo/rng.hpp]: Was this the cause of doxygen troubles??

File: src/detail/type_name.cpp [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+namespace pagmo
+{
+
+namespace detail
+{
+
+std::string demangle_from_typeid(const char *s)
+{
+#if defined(__GNUC__) || (defined(__clang__) && !defined(_MSC_VER))
```
Comment (darioizzo) [inline src/detail/type_name.cpp]: e che cazzo! ma mettersi daccordo no? in questo caso mi pare che MSVC faccia la cosa piu sana :)

File: doc/sphinx/install.rst (hunk 1 above)
Comment (bluescarni) [reply doc/sphinx/install.rst]: This was added recently to the conda-forge instructions on their website.

File: include/pagmo/algorithm.hpp (hunk 2 above)
Comment (bluescarni) [reply include/pagmo/algorithm.hpp]: Undefined behaviour of course :) This is just the standard warning that is there on all ``extract()`` functions.

File: include/pagmo/rng.hpp (hunk 3 above)
Comment (bluescarni) [reply include/pagmo/rng.hpp]: No idea, this is just the workaround for the current doxygen version (1.8.17). Other doxygen versions give errors in other places.
